worlds/hades/Client.py

```diff
@@ -0,0 +1,409 @@
+                
+        #Codify in the string all the fillers values
+        hades_settings_string += str(self.hades_slot_data["darkness_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["keys_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["gemstones_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["diamonds_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["titan_blood_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["nectar_pack_value"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["ambrosia_pack_value"]) + "-"
+
+        #Codify in the string all game settings
+        hades_settings_string += str(self.hades_slot_data["location_system"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["reverse_order_em"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["keepsakesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["weaponsanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["storesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["initial_weapon"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["ignore_greece_deaths"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["fatesanity"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["hidden_aspectsanity"]) + "-"
+        hades_settings_string += str(self.polycosmos_version) + "-"
+        hades_settings_string += str(self.hades_slot_data["automatic_rooms_finish_on_hades_defeat"]) + "-"
+
+        #Codify in the string all the finishing conditions
+        hades_settings_string += str(self.hades_slot_data["hades_defeats_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["weapons_clears_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["keepsakes_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["fates_needed"]) + "-"
+            
+        #Store the compact setting string
+        self.compact_setting_string = hades_settings_string
+
+    def request_location_to_item_dictionary(self):
+        self.creating_location_to_item_mapping = True
+        request = self.missing_locations_cache + self.checked_locations_cache
+        asyncio.create_task(self.send_msgs([{"cmd": "LocationScouts", "locations": request, "create_as_hint": 0}]))
+
+    async def create_location_to_item_dictionary(self, itemsdict):
+        locationItemMapping = ""
+        for networkitem in itemsdict:
+            locationItemMapping += self.clear_invalid_char(self.location_names.lookup_in_slot(networkitem.location)) \
+                + "--" + self.clear_invalid_char(self.player_names[networkitem.player]) + "--" \
+                + self.clear_invalid_char(self.item_names.lookup_in_slot(networkitem.item, networkitem.player)) + "||"
```

Comment:
Be prepared to explode if any player, item, or location name ever includes `--` or `||`. If you can come up with a way to safeguard this, you probably should.